utils/utils.go

```diff
@@ -256,9 +268,14 @@ func GeneratePasswd(uid, gid uint32, rootfs, rundir string) (string, error) {
 		}
 		return "", errors.Wrapf(err, "unable to read passwd file %s", originPasswdFile)
 	}
-
-	pwd := fmt.Sprintf("%s%d:x:%d:%d:container user:%s:/bin/sh\n", orig, uid, uid, gid, "/")
-	if err := ioutil.WriteFile(passwdFile, []byte(pwd), 0644); err != nil {
+	if username == "" {
+		username = "default"
+	}
+	if homedir == "" {
+		homedir = "/tmp"
+	}
+	pwd := fmt.Sprintf("%s%s:x:%d:%d:%s user:%s:/sbin/nologin\n", orig, username, uid, gid, username, homedir)
+	if err := ioutil.WriteFile(passwdFile, []byte(pwd), info.Mode()); err != nil {
```

Comment:
We need to chown the file to the original file's user.group.

Reply:
wait - one more question - why aren't we appending to the original file's contents?

Reply:
We are, that is what is in the orig variable.

Reply:
:+1: